apps/system/build/build.js

```diff
@@ -99,10 +100,29 @@ SystemAppBuilder.prototype.integrateLockScreenInputpad = function(options) {
   utils.writeContent(systemIndexFile, replacedIndexContent);
 };
 
+SystemAppBuilder.prototype.writeSearchPartnerJson = function() {
+
+  var variantFile = utils.getFile(this.distDirPath, 'variant.json');
+  var searchProviderJson =
+    utils.getJSON(utils.getFile(this.gaiaDirPath, 'shared', 'js',
+                                'search_providers_input.json'));
+
+  if (variantFile.exists()) {
+    var variantJson = utils.getJSON(variantFile);
+    if ('search_partner_code' in variantJson) {
+      searchProviderJson.partner_code = variantJson.search_partner_code;
+    }
+  }
+  var outFile = utils.getFile(this.gaiaDirPath, 'shared', 'js',
+                              'search_providers.json');
```

Comment:
@cctuan yeh it worked for me, but theres a good chance I was running a build that generated the file before flashing every time, will try to take a look at what contacts-import-server is doing or how to set up a new build module

@KevinGrandon The name of the generated file didnt change, just the file that was used to generate it, I didnt see any conventions we had around that